apps/ledger-live-desktop/src/renderer/components/CustomImage/ImageCropper.tsx

```diff
@@ -0,0 +1,244 @@
+  ctx.rotate(rotRad);
+  ctx.translate(-image.naturalWidth / 2, -image.naturalHeight / 2);
+
+  // draw rotated image
+  ctx.drawImage(image, 0, 0);
+
+  // croppedAreaPixels values are bounding box relative
+  // extract the cropped image using these values
+  const data = ctx.getImageData(pixelCrop.x, pixelCrop.y, pixelCrop.width, pixelCrop.height);
+
+  // set canvas width to final desired crop size - this will clear existing context
+  canvas.width = pixelCrop.width;
+  canvas.height = pixelCrop.height;
+
+  // paste generated rotate image at the top left corner
+  ctx.putImageData(data, 0, 0);
+
+  // new canvas used to resize to the target dimensions
+  const { context: resizedCtx, canvas: resizedCanvas } = createCanvas();
+  if (resizedCtx === null) throw new Error('Context "croppedCtx" is null');
+  resizedCanvas.width = targetDimensions.width;
+  resizedCanvas.height = targetDimensions.height;
+  resizedCtx.drawImage(
+    canvas,
+    0,
+    0,
+    pixelCrop.width,
+    pixelCrop.height,
+    0,
+    0,
+    targetDimensions.width,
+    targetDimensions.height,
+  );
+
+  return {
+    imageBase64DataUri: resizedCanvas.toDataURL(),
+    height: resizedCanvas.height,
+    width: resizedCanvas.width,
+  };
+}
+
+const ImageCropper: React.FC<Props> = props => {
+  const {
+    imageBase64DataUri,
+    targetDimensions,
+    onResult,
+    onError,
+    initialCropParams,
+    setCropParams,
+    setLoading,
+  } = props;
+
+  const [crop, setCrop] = useState({ x: 0, y: 0 });
+  const [completeCropPixel, setCompleteCropPixel] = useState<Crop>();
+
+  const imageUuid = imageBase64DataUri;
+  const shouldUseInitialCropParams = initialCropParams && initialCropParams.imageUuid === imageUuid;
+  const [initialCroppedAreaPixels] = useState(
+    shouldUseInitialCropParams ? initialCropParams?.crop : undefined,
+  );
+  const [rotationIncrements, setRotationIncrements] = useState<number>(
+    shouldUseInitialCropParams ? initialCropParams.rotationIncrements : 0,
+  );
+  const [zoom, setZoom] = React.useState(shouldUseInitialCropParams ? initialCropParams.zoom : 1);
+  const imageRef = useRef<HTMLImageElement | null>(null);
+  const aspect = targetDimensions.width / targetDimensions.height;
+  const debouncedCompleteCropPixel = useDebounce(completeCropPixel, 500);
+
+  useEffect(() => {
+    let dead = false;
+    if (!imageRef.current) return;
+    if (!imageRef.current.complete) return;
+    if (!debouncedCompleteCropPixel) return;
+    setLoading(true);
+    cropAndResizeImage(
+      imageRef.current,
+      targetDimensions,
+      debouncedCompleteCropPixel,
+      rotationIncrements,
+    )
+      .then(res => {
+        setLoading(false);
+        if (dead) return;
+        onResult(res);
+      })
+      .catch(e => {
+        if (dead) return;
+        console.error(e);
+        onError(new ImageCropError());
+      });
+    return () => {
+      dead = true;
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
maybe add a comment to explain why some variables should not be in the dependency array